### PR TITLE
Fix Deprecated code

### PR DESCRIPTION
**Pull request by Lukapetro**
This PR addresses the issue of deprecated code in `app/index.tsx` and `_components/posts.tsx`.

The update involve a refactor from `api.useContext` to `api.useUtils` -> [docs](https://trpc.io/docs/client/react/useUtils).

These changes ensure compatibility with the latest version of `TRPc`.

### CODE REVIEW

File: apps/nextjs/src/app/_components/posts.tsx [hunk 1]
```diff
@@ -2,11 +2,11 @@
 
 import { useState } from "react";
 
-import { api } from "~/utils/api";
 import type { RouterOutputs } from "~/utils/api";
+import { api } from "~/utils/api";
 
 export function CreatePostForm() {
-  const context = api.useContext();
+  const context = api.useUtils();
```
Comment (juliusmarminge) [inline apps/nextjs/src/app/_components/posts.tsx:9]: utils =

File: apps/nextjs/src/app/_components/posts.tsx [hunk 2]
```diff
@@ -99,7 +99,7 @@ export function PostList() {
 export function PostCard(props: {
   post: RouterOutputs["post"]["all"][number];
 }) {
-  const context = api.useContext();
+  const context = api.useUtils();
```
Comment (juliusmarminge) [inline apps/nextjs/src/app/_components/posts.tsx:102]: utils =

**Review by juliusmarminge — CHANGES_REQUESTED** [review]
👍

File: apps/nextjs/src/app/_components/posts.tsx (hunk 1 above)
Comment (Lukapetro) [reply apps/nextjs/src/app/_components/posts.tsx]: fixed

File: apps/nextjs/src/app/_components/posts.tsx (hunk 2 above)
Comment (Lukapetro) [reply apps/nextjs/src/app/_components/posts.tsx]: fixed

**Review by juliusmarminge — APPROVED** [review]

**Pull request merged**